### PR TITLE
Remove request.schema

### PR DESCRIPTION
**Pull request by amaltaro**
Given that this request.schema section isn't updated during the assignment - thus it might contain inconsistent info - and that reqmgr2 json provides all those same info, we decided to get rid of this legacy code.

Changes are:
* removed the legacy request.schema section from that page that we dump the workload properties, e.g.: https://cmsweb.cern.ch/reqmgr2/config?name=pdmvserv_task_HIG-RunIIFall17wmLHEGS-00310__v1_T_180115_163848_4687
* set dbsUrl at workload level (still look up at the old request.schema for old requests)
* updated the workload getRequestType as well (keeping old format too)

### CODE REVIEW

File: src/python/WMCore/ReqMgr/ExternalResourceTool.py [hunk 1]
```diff
@@ -8,7 +8,7 @@ def handlePreSideEffect(workloadHelper, request_args):
     # handle some thing other than standard procedure of update
     # before the REQMGR rest api succeeded
     # currently just handling store result assigment-approve and assigned
-    if workloadHelper.requestType() == "StoreResult":
+    if workloadHelper.getRequestType() == "StoreResult":
```
Comment (amaltaro) [inline src/python/WMCore/ReqMgr/ExternalResourceTool.py]: buggy, it should be "StoreResults". Same for code below and WMStats package.

**Comment by vlimant** [comment]
@amaltaro what does this do in practice ?

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
This is supposed to remove the _legacy_ `request.schema` section from the dump of the workload. That information can be found (and AFAIK it's somehow still used) in
https://cmsweb.cern.ch/reqmgr2/config?name=fabozzi_Run2016H-03Feb2017_ver3-v1-SingleMuon-05Jan2018_ver3_942_180108_122210_3864

which contains the original request schema settings (plus some default) and sometimes some inconsistent info as well.

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
@ticoann I'm still running a few tests, but can you please start reviewing it?

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
I tested central services and the agent with and without this patch and I couldn't spot any issues. So I don't think there will be any backwards incompatibility when we push it in.

@ticoann I think I can ask for your review now.

**Comment by ticoann** [comment]
Patch looks good, merging it.

**Review by ticoann — APPROVED** [review]

**Comment by ticoann** [comment]
The only thing is I am not sure whether reqmgr2 GUI is affected by this. We will see.

**Comment by amaltaro** [comment]
I did play a bit with the reqmgr2 gui, also created an ACDC via wmstats and assigned it via web and had no issues. But we can test it further in testbed.

**Pull request merged**